src/app/qgisapp.cpp

```diff
@@ -6204,36 +6212,73 @@ void QgisApp::toggleFullScreen()
 }
 
 void QgisApp::togglePanelsVisibility()
+{
+  toggleReducedView( false );
+}
+
+void QgisApp::toggleMapOnly()
+{
+  toggleReducedView( true );
+}
+
+void QgisApp::toggleReducedView( bool viewMapOnly )
 {
   QgsSettings settings;
 
   QStringList docksTitle = settings.value( QStringLiteral( "UI/hiddenDocksTitle" ), QStringList() ).toStringList();
   QStringList docksActive = settings.value( QStringLiteral( "UI/hiddenDocksActive" ), QStringList() ).toStringList();
+  QStringList toolBarsActive = settings.value( QStringLiteral( "UI/hiddenToolBarsActive" ), QStringList() ).toStringList();
 
   QList<QDockWidget *> docks = findChildren<QDockWidget *>();
   QList<QTabBar *> tabBars = findChildren<QTabBar *>();
+  QList<QToolBar *> toolBars = findChildren<QToolBar *>();
 
-  if ( docksTitle.isEmpty() )
+  bool allWidgetsVisible = settings.value( QStringLiteral( "UI/allWidgetsVisible" ), true ).toBool();
+
+  if ( allWidgetsVisible )  // that is: currently nothing is hidden
   {
 
+    if ( viewMapOnly )  //
+    {
+      // hide also statusbar and menubar and all toolbars
+      Q_FOREACH ( QToolBar *toolBar, toolBars )
```

Comment:
Remove the QFOREACH?
https://github.com/qgis/QGIS-Enhancement-Proposals/issues/103
Same for line 6302

Reply:
@Gustry sure, QEP is decided in favour of removing? 
Will try to do this later, but could need some help if in trouble... (note that this was excisting code)